src/encoded/static/components/genome_browser.js

```diff
@@ -522,7 +522,22 @@ class GenomeBrowser extends React.Component {
         if (files.length > 0) {
             tracks = this.filesToTracks(newFiles, this.props.label, domain);
         }
-        this.setState({ trackList: tracks }, () => {
+        let contig = this.state.contig;
+        let x0 = this.state.x0;
+        let x1 = this.state.x1;
+        if (this.chartdisplay) {
+            const scrollLocation = this.chartdisplay.getElementsByClassName('hpgv_panel-header')[0].getElementsByTagName('span')[0].innerText;
```

Comment:
The older means of referencing DOM like using "getElementsByClassName" is done when you need I speed. The speed difference is negligible unless you are using it in a loop with tens of thousands of iterations. Is there a special reason you are using it?

Reply:
Do you mean that it would be better to use a ref? I think that could be tricky since I'm referring to an element generated in a component in a library. I thought about making a method to return the browser position but this was the simplest solution since it didn't require any changes to Valis and I think it should work consistently.